src/main/kotlin/no/nav/helsearbeidsgiver/tokenprovider/RestSTSAccessTokenProvider.kt

```diff
@@ -35,25 +36,28 @@ class RestSTSAccessTokenProvider(
     init {
         basicAuth = basicAuth(username, password)
         endpointURI = "$stsEndpoint?grant_type=client_credentials&scope=openid"
-        currentToken = requestToken()
+        currentToken = runBlocking { requestToken() }
     }
 
     override fun getToken(): String {
         if (isExpired(currentToken, Date.from(Instant.now().plusSeconds(300)))) {
             log.debug("OIDC Token is expired, getting a new one from the STS")
-            currentToken = requestToken()
+            currentToken = runBlocking { requestToken() }
             log.debug("Hentet nytt token fra sts som går ut ${currentToken.expirationTime}")
         }
         return currentToken.tokenAsString
     }
 
-    private fun requestToken(): JwtToken {
+    private suspend fun requestToken(): JwtToken {
         val response = runBlocking {
-            httpClient.get<STSOidcResponse>(endpointURI) {
-                headers.append("Authorization", basicAuth)
-                headers.append("Accept", "application/json")
+            httpClient.get(endpointURI) {
+                headers {
+                    append("Authorization", basicAuth)
+                    append("Accept", "application/json")
+                }
```

Comment:
Ktor 2 har en egen funksjon for autorisasjon. Tror det er `basicAuth(...)`. På den andre kan du bruke `header(HttpHeaders.Accept, ContentType.Application.Json.toString())` (er ikke fan av strings).